### PR TITLE
Add axis types as chart operator flags

### PR DESCRIPTION
**Pull request by dit7ya**
Relevant for https://github.com/tenzir/issues/issues/1256.

### CODE REVIEW

File: libtenzir/builtins/operators/chart.cpp [hunk 1]
```diff
@@ -724,10 +724,14 @@ chart_definition chart_definitions[] = {
       {.attr = "x", .flag = "-x,--x-axis", .type = flag_type::field_name, .default_ = nth_field{0}, .allow_lists = false, .req = requirement::strictly_increasing,},
       {.attr = "y", .flag = "-y,--y-axis", .type = flag_type::field_name, .default_ = nth_field{1, nth_field::all_the_rest}, .allow_lists = true,},
       {.attr = "position", .flag = "--position", .type = flag_type::attribute_value, .default_ = attribute_value{"grouped"}, .allow_lists = false,},
+      {.attr = "xAxisType", .flag = "--x-axis-type", .type = flag_type::attribute_value, .default_ = attribute_value{"linear"}, .allow_lists = false,},
+      {.attr = "yAxisType", .flag = "--y-axis-type", .type = flag_type::attribute_value, .default_ = attribute_value{"linear"}, .allow_lists = false,},
```
[the remark this answers is not in the document]
Comment (mavam) [reply libtenzir/builtins/operators/chart.cpp:728]: I like `--log-x` as boolean as well.

(I don't think the `--x-axis-type is too long, though, but no strong opinion.)

File: web/docs/operators/chart.md [hunk 2]
```diff
@@ -45,6 +45,18 @@ Control how the values are grouped when rendered as a chart.
 Possible values are `grouped` and `stacked`.
 Defaults to `grouped`.
 
+### `x-axis-type=<x-axis-type>` (`line`, `area`, and `bar` charts only)
+
+Sets the x-axis scale type.
+Possible values are `linear` and `log`.
+Defaults to `linear`.
+
+### `y-axis-type=<y-axis-type>` (`line`, `area`, and `bar` charts only)
```
Comment (dominiklohmann) [inline web/docs/operators/chart.md:54]:
```suggestion
### `--y-axis-type=<y-axis-type>` (`line`, `area`, and `bar` charts only)
```

(same above for y-axis-type and position)

File: web/docs/operators/chart.md [hunk 3]
```diff
@@ -45,6 +45,18 @@ Control how the values are grouped when rendered as a chart.
 Possible values are `grouped` and `stacked`.
 Defaults to `grouped`.
 
+### `x-axis-type=<x-axis-type>` (`line`, `area`, and `bar` charts only)
+
+Sets the x-axis scale type.
+Possible values are `linear` and `log`.
+Defaults to `linear`.
```
Comment (mavam) [inline web/docs/operators/chart.md:52]:
```suggestion
Possible values are `linear` and `log`.

Defaults to `linear`.
```

File: libtenzir/builtins/operators/chart.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (dit7ya) [reply libtenzir/builtins/operators/chart.cpp:728]: We might want to add other axis-types in future beyond log (for example - https://d3js.org/d3-scale), so I wouldn't prefer `--x-log`, and I think even though `--x-axis-type` is a bit long, it is a bit more clear, although not super strong opinion.

File: web/docs/operators/chart.md (hunk 3 above)
Comment (dit7ya) [reply web/docs/operators/chart.md:52]: The same style has too many other instances (without the newline) in the file and I don't feel like it adds much value.

Comment (mavam) [reply web/docs/operators/chart.md:52]: Can you fix all of them. All other operators have that convention about putting the defaults into a dedicated line.

File: libtenzir/builtins/operators/chart.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (jachris) [reply libtenzir/builtins/operators/chart.cpp]: Sure. We'll have the chance to chance this again with TQL2 anyway, so whatever work's for now.

**Review by jachris — APPROVED** [review]

**Review by mavam — APPROVED** [review]
I didn't test this, but existing DSL makes it easy to review as diff. @dit7ya if it works for you, green light.

File: web/docs/operators/chart.md (hunk 3 above)
Comment (dit7ya) [reply web/docs/operators/chart.md]: Okay done.

**Comment by mavam** [comment]
Our Markdown linter isn't happy yet:

```
web/docs/operators/chart.md:28:35 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
```

File: web/docs/operators/chart.md [hunk 4]
```diff
@@ -12,8 +12,11 @@ Add metadata to a schema, necessary for rendering as a chart.
 
 ```
 chart line [-x|--x-axis <fields>] [-y|--y-axis <field>]
+           [--x-axis-type=<x-axis-type>]  [--y-axis-type=<y-axis-type>]
 chart area [-x|--x-axis <fields>] [-y|--y-axis <field>]
+           [--x-axis-type=<x-axis-type>]  [--y-axis-type=<y-axis-type>]
 chart bar  [-x|--x-axis <fields>] [-y|--y-axis <field>]
+           [--x-axis-type=<x-axis-type>]  [--y-axis-type=<y-axis-type>]
 chart pie  [--name <field>] [--value <fields>]
```
Comment (dominiklohmann) [inline web/docs/operators/chart.md:20]: Inconsistency nitpick: we generally don't have an equal sign after long options in the docs.

This also applies to the newly added example at the bottom.

Comment (mavam) [reply web/docs/operators/chart.md]: Thanks for the 🦅 👀!

**Pull request merged**